spec/traverse_sequences_spec.rb

```diff
@@ -127,13 +127,11 @@
     end
 
     it 'has done markers differently-sized sequences' do
-      expect(@callback_s1_s2.done_a).to eq([['p', 9, 's', 10]])
+      expect(@callback_s1_s2.done_a).to eq([['p', 9, 't', 11]])
       expect(@callback_s1_s2.done_b).to be_empty
 
-      # 20110731 I don't yet understand why this particular behaviour
-      # isn't transitive.
       expect(@callback_s2_s1.done_a).to be_empty
-      expect(@callback_s2_s1.done_b).to be_empty
+      expect(@callback_s2_s1.done_b).to eq([['t', 11, 'p', 9]])
```

Comment:
Once we fixed the off-by-one error, the behavior is transitive, so I think the comment at line 133 is no longer necessary.

P/S: It's a bit off-topic, but if this fix is correct, I find that comment is one of the best ones I've ever seen, it's amazing how it helps me understand the situation, more than 10 years later :).

Reply:
Thank you. And I think that your fix is correct. Will be merging it.